specification/maps/data-plane/DataRegistry/stable/2022-12-01/dataregistry.json

```diff
@@ -0,0 +1,712 @@
+          {
+            "value": "geojson",
+            "description": "[GeoJSON](https://tools.ietf.org/html/rfc7946) is a JSON based geospatial data interchange format."
+          },
+          {
+            "value": "zip",
+            "description": "Compressed data format."
+          }
+        ]
+      },
+      "x-ms-parameter-location": "method"
+    },
+    "UniqueDataId": {
+      "name": "udid",
+      "description": "The unique data id for the data registry. The `udid` must have been obtained from a successful [Data Upload API](https://docs.microsoft.com/en-us/rest/api/maps/data-v2/upload-preview) call.",
+      "type": "string",
+      "in": "path",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    },
+    "DataDescription": {
+      "name": "description",
+      "description": "The description to be given to the data registry.",
+      "type": "string",
+      "in": "query",
+      "x-ms-parameter-location": "method"
+    },
+    "OperationId": {
+      "name": "operationId",
+      "type": "string",
+      "in": "path",
+      "description": "The ID to query the status for the data upload request.",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    }
+  },
+  "paths": {
+    "/mapData": {
+      "post": {
+        "description": "***Applies to:** see pricing [tiers](https://aka.ms/AzureMapsPricingTier).\n\nThe DataRegistry Create API allows the caller to create a data registry for their data content uploaded to Azure Storage Account in the Azure Maps Service and in exchange of a UDID. You can use this API in a scenario like registering a collection of Geofences in `GeoJSON` format, for use in our [Azure Maps Geofencing Service](https://docs.microsoft.com/rest/api/maps/spatial).\n\n## Submit Create Request\n\nTo create a data registry you will use a `POST` request. The request body will contain the connection object. The `dataFormat` query parameter will contain the format for the data for validation.\n\nFor example, to upload a collection of geofences  in `GeoJSON` format, set the request body to the connection object. Set the `dataFormat` query parameter to _geojson_.\n\nThe DataRegistry Create API performs a [long-running request](https://aka.ms/am-creator-lrt-v2).",
+        "operationId": "DataRegistries_Create",
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "x-ms-examples": {
+          "Register GeoJSON data containing geometries that represent a collection of geofences to the Azure Maps Service": {
+            "$ref": "./examples/DataRegistries_Create.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../Common/preview/1.0/common.json#/parameters/ClientId"
+          },
+          {
+            "$ref": "#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "#/parameters/DataFormat"
```

Comment:
should we move `description` and `dataFormat` to request body?
Data V2 put them in query parameter because the request body of Data V2 is pure file content.
Given that `description` and `dataFormat` are part of the data metadata, it's better to merge into request body.

Reply:
agree!